main-presentation/src/main/java/com/teamhy2/feature/home/HomeViewModel.kt

```diff
@@ -37,72 +33,63 @@ class HomeViewModel
 
         init {
             loadPromotionData()
             loadHomeData()
             updateDeviceToken()
         }
 
-        private fun loadHomeData() {
-            viewModelScope.launch {
+        private fun loadHomeData() =
+            intent {
                 runCatching {
-                    listOf(
-                        async { wiseSayingRepository.fetchWiseSaying().getOrThrow() },
-                        async { studyDayRepository.fetchWeeklyStudyDay().getOrThrow() },
-                    ).awaitAll()
-                }.onSuccess { results ->
-                    val (wiseSaying, weeklyStudyDays) = results
+                    coroutineScope {
+                        val deferredWiseSaying =
+                            async { wiseSayingRepository.fetchWiseSaying().getOrThrow() }
+                        val deferredWeeklyStudyDays =
+                            async { studyDayRepository.fetchWeeklyStudyDay().getOrThrow() }
 
-                    _homeUiState.update {
-                        HomeUiState.Success(
-                            wiseSaying = wiseSaying as WiseSaying,
-                            weeklyStudyDays = weeklyStudyDays as List<WeeklyStudyDay>,
-                        )
+                        deferredWiseSaying.await() to deferredWeeklyStudyDays.await()
                     }
-                }.onFailure { exception ->
-                    _errorFlow.emit(exception)
-                    _homeUiState.value = HomeUiState.Error(exception.message)
                 }
+                    .onSuccess { (wiseSaying, weeklyStudyDays) ->
+                        reduce {
+                            HomeState.Success(
+                                wiseSaying = wiseSaying,
+                                weeklyStudyDays = weeklyStudyDays,
+                            )
+                        }
+                    }
+                    .onFailure { postSideEffect(HomeSideEffect.ShowError(it)) }
             }
-        }
 
-        private fun loadPromotionData() {
-            viewModelScope.launch {
+        private fun loadPromotionData() =
+            intent {
                 runCatching {
                     val promotion: Promotion = promotionRepository.fetchPromotionData().getOrThrow()
                     val isDismissedFlow: Flow<Boolean> = promotionRepository.isPromotionDismissed
 
                     isDismissedFlow.collect { isDismissed ->
-                        _homeUiState.update { currentState ->
-                            if (currentState is HomeUiState.Success) {
-                                currentState.copy(
+                        runOn<HomeState.Success> {
+                            reduce {
+                                state.copy(
                                     promotion = promotion,
                                     isPromotionDialog = !isDismissed && promotion.isActive,
```

Comment:
```suggestion
                                    isPromotionDialog = isDismissed.not() && promotion.isActive,
```
여기도 일관성 있게 not이 붙으면 좋겠네요!